File: MongoButcher/App/Core/Workloads/Recipes/Recipe.cs
using System;
using System.Collections.Generic;
using LeoMongo.Database;
using MongoDBDemoApp.Core.Workloads.Products;
using MongoDBDemoApp.Core.Workloads.Resources;

namespace MongoDBDemoApp.Core.Workloads.Recipes
{
    public class Recipe : EntityBase
    {
        public String Name { get; set; }
        public String Procedure { get; set; }
        public Product Endproduct { get; set; }
        public List<Resource> Incrediants { get; set; }
    }
}

File: MongoButcher/Client/src/app/data/recipe.ts
import {Product} from './product';
import {Resource} from './resource';

export class Recipe {
  id ?: string;
  name ?: string;
  procedure ?: string;
  endProduct ?: Product;
  incrediants ?: Array<Resource>;

  constructor(id?: string, name?: string, procedure?: string, output?: Product, ingredients?: Array<Resource>) {
    this.id = id;
    this.name = name;
    this.procedure = procedure;
    this.incrediants = ingredients;
    this.endProduct = output;
  }
}
